### PR TITLE
Update archlinux building documentation, and delete obsolete part

### PR DESCRIPTION
**Pull request by neowutran**
Archlinux specific documentation: 
- /doc/templates/archlinux/
- /doc/building-archlinux-template/

Both of theses documentation are outdated. 

/doc/templates/archlinux/
--------------------------------

* Everything is about Qubes 3.2 and below
* Olivier Medoc no longer maintain the template
* So the binary packages don't exist anymore
* So the prebuild template don't exist anymore

Maybe it is better to keep the archlinux template as "build-only", so I suggest to just delete the whole page. If this point of view is shared by the majority and is accepted, change should be made in the source code to disable the build of the keyring package, and remove the gpg keys from it. 
If the point of view of the majority is to keep the binary packages / prebuild template, the code should still be changed to update the gpg key of the new maintainer. 

/doc/building-archlinux-template/
-------------------------------

* Everything is about Qubes 3.2 and below

Updated the building process in itself, removed the obsolete "Known problems", removed the obsolete "Qubes mailing list", added some documentation on how to debug the build process and how to debug the qube runtime, and added basic information about how to setup a archlinux repository to host the packages updates

Dependencies
----------------------

Some screenshot were deleted / added / updated. Once this pull request have reached a "ready to merge" status, the qubes-attachment should be updated with the new screenshot
https://github.com/neowutran/qubes-attachment

### CODE REVIEW

**Comment by andrewdavidwong** [comment]
@marmarek, if there's a more appropriate reviewer for this, please feel free to reassign.

File: external/building-guides/building-archlinux-template.md [hunk 1]
```diff
@@ -8,484 +8,437 @@ redirect_from:
...
-
+```shell_session
+$ gpg --import /usr/share/qubes/qubes-master-key.asc
+```
 * Verify its fingerprint, set as 'trusted'. [This is described here](/doc/VerifyingSignatures).
-
-* Download the Qubes developers' keys.
-
-      wget https://keys.qubes-os.org/keys/qubes-developers-keys.asc
-      gpg --import qubes-developers-keys.asc
```
Comment (marmarek) [inline external/building-guides/building-archlinux-template.md]: This step is necessary, just master key isn't enough.

Comment (neowutran) [reply external/building-guides/building-archlinux-template.md]: Indeed, thanks

**Comment by marmarek** [comment]
Automatic spellchecker reported unknown words:
```
    +09a435fcc6bdcb19144d198ea20f7a27826c1d80
    +Exemple
    +archlinux2
    +cmp
    +diffutils
    +exemple
    +hotfix
    +neowutran
    +release4
    +rwdev
    +tryied
    +url
    +xf86dgaproto
```
Some are typos to be fixed, some looks like code not properly marked as such.

**Pull request merged**